lib/ecto/adapters/sql.ex

```diff
@@ -303,7 +317,7 @@ defmodule Ecto.Adapters.SQL do
   @spec rollback_test_transaction(Ecto.Repo.t, Keyword.t) :: :ok
   def rollback_test_transaction(repo, opts \\ []) do
     pool = pool!(repo)
```

Comment:
I was thinking that instead of calling repo.__pool__ twice, we should make `pool!(repo)` return the repo and timeout. What do you think?